gflownet/gflownet.py

```diff
@@ -898,15 +899,17 @@ def test_top_k(self, it, progress=False, gfn_states=None, random_states=None):
         print()
         if not gfn_states:
             # sample states from the current gfn
+            batch = Batch(env=self.env, device=self.device, float_type=self.float)
             self.random_action_prob = 0
-            gfn_states = []
             t = time.time()
             print("Sampling from GFN...", end="\r")
-            for b in batch_with_rest(0, self.logger.test.n_top_k, self.batch_size):
-                gfn_states += self.sample_batch(
-                    self.env, len(b), train=False, progress=progress
-                )[0]
+            for b in batch_with_rest(
+                0, self.logger.test.n_top_k, self.batch_size_total
+            ):
+                sub_batch, _ = self.sample_batch(n_forward=len(b), train=False)
+                batch.merge(sub_batch)
             duration = time.time() - t
+            gfn_states = batch.get_terminating_states()
```

Comment:
For my own understanding : 
Is this change (and the following one a few lines down) merely for code consistency (using the new Batch class) or does it also solve a bug at the same time?

Reply:
The previous way simply didn't work (it did work when it was implemented by Victor, so not strictly a bug) because the interface of `gflownet.sample_batch()` changed with the Batch v2.